package.json

```diff
@@ -16,6 +16,13 @@
 	"description": "An ESnext spec-compliant `Array.prototype.flat` shim/polyfill/replacement that works as far down as ES3.",
 	"license": "MIT",
 	"main": "index.js",
+	"files": [
+		"index.js",
```

Comment:
The “files” field is dangerous - the only way published files should ever be restricted is by npmignore.

Reply:
There is an interesting discussion on the topic here: https://medium.com/@jdxcode/for-the-love-of-god-dont-use-npmignore-f93c08909d8d

Why is the `files` filed dangerous?

Reply:
Omitting a file breaks your consumers, which is far worse than the cost of including an extra file on npmignore (an insignificant number of useless bytes in a package)

Reply:
Also, that article is ridiculous, because if your credentials ever exist on a file inside your project, you’ve already lost security - any postinstall script can read it, for example.

Leaked credentials can be changed. Broken consumers will forever remember the pain you’ve inflicted on them.

Reply:
Fair point about the credentials, I still quite like whitelisting though as it's very explicit what is being published!

I agree with you though - one should never break consumers!

Would you say that `.eslintrc`, `.editorconfig`, `.travis.yml`, `.nycrc`, `.editorconfig` are necessary when publishing to NPM? Or is it too late do ignore these files for such a well-used project like this, as someone might be using these files after installing it?

Reply:
editorconfig and eslintrc and nycrc are, since tests should be verifying style.

You could exclude the travis config, basically, and not much else - which is such a tiny file that it isn’t really worth it.